internal/controller/controllers/bmh_agent_controller.go

```diff
@@ -847,6 +852,7 @@ func (r *BMACReconciler) findBMHByAgent(ctx context.Context, agent *aiv1beta1.Ag
 			}
 		}
 	}
+	log.Warn("Did not find BareMetalHost corresponding to the Agent")
```

Comment:
`findBMHByAgent` is used for the watchers to know whether a BMH request should be queued when there's an event for an agent. 

I think this has the potential to flood the logs without helping with the user experience.  As it is now, it will log a message as soon as the BMH is created and for every agent that we have events for.

As far as the user experience goes, it would be more useful to have an event on a specific BMH. I [wrote a comment on the BZ associated with this PR](https://bugzilla.redhat.com/show_bug.cgi?id=1969975#c1) explaining my reasoning here.